### DISCUSSION TITLE
How to have image fill width yet not be squished in height using next/image?

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#45774] [Help]
I want to have a responsive page with a sidebar on left and right and an image in the center which fills the width. The sidebars are always 300px wide, but the center is a dynamic width, with a dynamic width blog post containing dynamic width images. How can I use `next/image` and obtain the optimizations discussed in https://nextjs.org/docs/messages/no-img-element ? If I set `fill` on the image, it squishes the image when I open the chrome debugger, i.e. when the height is different, so the aspect ratio gets messed up. I just want to resort back to the plain `<img>` tag, but perhaps there is another way to gain the benefits described in the [Next.js image docs](https://nextjs.org/docs/api-reference/next/image#blurdataurl)?

Maybe there is a way to just `zoom` in CSS to a specific aspect ratio to fit the image or some hack?

Is doing this correct?

```jsx
<Image
  style={{ height: 'auto', width: '100%' }}
  alt={alt}
  blurDataURL={preview}
  width={w}
  height={h}
  src={actualSrc}
/>
```

**Comment by ciruz** [comment]
Hey, does my previous answer maybe helps you too?

With next.js ``13.x`` you can now simply use the ``fill`` attribute, like so:

```javascript
<Image src={image} alt="" fill />
```

result:
![image](https://user-images.githubusercontent.com/987947/209806162-69db51a9-6229-44fb-8631-4301007fcfb0.png)


**If you want to style it, the ``parent`` element must have ``position: relative`` and ``display: block``.**

If you want to control the size of the image, you use the parent element with fill, like so:
```javascript
<div
  style={{
    position: "relative",
    width: "300px",
    height: "100px",
    border: "solid 3px blue",
  }}
>
    <Image src={image} alt="" fill />
</div>
```

result:
![image](https://user-images.githubusercontent.com/987947/209806616-9df49be7-87f0-4de6-b553-81e62ab6cb4d.png)


Or if this is not the desired look, use ``object-fit`` now via ``css/style`` instead of an attribute, like it was the case with next 12.

```javascript
<div
  style={{
    position: "relative",
    width: "300px",
    height: "100px",
    border: "solid 3px blue",
  }}
>
    <Image src={image} alt="" fill style={{ objectFit: "cover" }} />
</div>
```

result:
![image](https://user-images.githubusercontent.com/987947/209806834-aea14a83-07ae-4833-bda6-e6e022fff0db.png)
